cmd/os-readdir_other.go

```diff
@@ -83,7 +83,7 @@ func readDirN(dirPath string, count int) (entries []string, err error) {
 			}
 			return nil, osErrToFileErr(err)
 		}
-		if count > 0 {
+		if count > -1 {
```

Comment:
This is nice but really we have no requirement where we want to list no entries 0, that's why the function was on purpose treating -1 and 0 the same..

Reply:
@harshavardhana The problem is that this causes the unit tests to fail when run on an OS that is `other` (in my case, solaris).  As far as I can tell, only `plan9` and `solaris` build with these, and clearly solaris isn't tested by the automated test suites. I'm not sure if `plan9` is, but it should fail as well.

For the specific test case, take a look in os-readdir_test.go in `TestReadDirN` and the 3rd test has `numFiles: 1, n: 0, expectedNum: 0`.

Reply:
> @harshavardhana The problem is that this causes the unit tests to fail when run on an OS that is `other` (in my case, solaris).  As far as I can tell, only `plan9` and `solaris` build with these, and clearly solaris isn't tested by the automated test suites. I'm not sure if `plan9` is, but it should fail as well.
> 
> For the specific test case, take a look in os-readdir_test.go in `TestReadDirN` and the 3rd test has `numFiles: 1, n: 0, expectedNum: 0`.
> 

This might be a different issue I don't it's related to this. The test cases pass fine on all Linux, macOS and BSDs as per my testing.

So something else is wrong. Changing the count is perhaps masking.

Reply:
@harshavardhana Yes, Linux, macOS and BSDs will pass, because they all use os-readdir_unix (`+build linux,!appengine darwin freebsd netbsd openbsd`), the os-readdir_other only applies to `plan9` and `solaris` (`+build plan9 solaris`).

The unix version (`os-readdir_unix`) explicitly returns from `readDirN` if count == 0 (the `for count!=0` falls through the first time, resulting in no return value). 

The windows version (`os-readdir_windows`) contains the same code to fall through if count == 0 (`for count!=0`).

The "other" version (`os-readdir_other`), which is only active under `plan9` and `solaris`, has a `for !done`) control loop instead, and an explicit check for `count > 0` in the previous version, which made `count == 0` and `count == -1` perform identically.

I don't think this is masking anything other than an interesting choice in the original implementation.

Reply:
Understood looks good.